node/src/Pos/Wallet/Web/ClientTypes/Functions.hs

```diff
@@ -175,12 +176,13 @@ countVotes = foldl' counter (0, 0)
                               else (n, m + 1)
 
 -- | Creates 'CTUpdateInfo' from 'ConfirmedProposalState'
-toCUpdateInfo :: ConfirmedProposalState -> CUpdateInfo
-toCUpdateInfo ConfirmedProposalState {..} =
+toCUpdateInfo :: BlockVersionData -> ConfirmedProposalState -> CUpdateInfo
+toCUpdateInfo bvd ConfirmedProposalState {..} =
     let UnsafeUpdateProposal {..} = cpsUpdateProposal
         cuiSoftwareVersion  = upSoftwareVersion
         cuiBlockVesion      = upBlockVersion
-        cuiScriptVersion    = bvmScriptVersion upBlockVersionMod
+        cuiScriptVersion    = maybe identity const (bvmScriptVersion upBlockVersionMod) $
+            bvdScriptVersion bvd
```

Comment:
Is this not just `fromMaybe (bvdScriptVersion bvd) (bvmScriptVersion upBlockVersionMod)` ?